Crisan Cristian Alexandru/Casualwood/rws.c

```diff
@@ -0,0 +1,71 @@
+#include <stdlib.h>
+#include <stdio.h>
+#include <conio.h>
+#include <string.h>
+#include <stdbool.h>
+typedef struct
+{
+    char *name;
+    char **friends;
+    int nrOfFriends;
+} actor;
+typedef struct
+{
+    char *name;
+    char **cast;
+    int CastSize;
+} movie;
+bool isUnique(actor *actors,int nrOfDActors,char *wantedActorName)
+{
+    for(int i=0; i<nrOfDActors; i++)
+    {
+        if(strcmp(actors[i].name,wantedActorName)==0)
+        {
+            return false;
+        }
+    }
+    return true;
+}
+void rasFILE(FILE *f,movie *movies,actor *actors,int *nrOfDactors,int *nrOfDMovies)
+{
+    *nrOfDactors=0;
+    *nrOfDMovies=0;
+    char *currentLine=(char*)malloc(sizeof(char)*200);
+    while((currentLine=fgets(currentLine,200,f))!=NULL)
+    {
+        printf("da");
+        movies[*nrOfDMovies].CastSize=0;
+        char *info=strtok(currentLine,",");
+        printf("da2");
+        strcpy(movies[*nrOfDMovies].name,info);
```

Comment:
Problema ta e de fapt, aici - dacă ai fi mers cu debuggerul, ai fi văzut că de fapt aici crapă, nu mai jos.
Motivul e simplu - tu ai un array de filme, `movies`. Fiecare `movie` are un `char * name` pe lângă alte cele. Ideea e că, degeaba în funcția main ai declarat tu:
`movie *movies=(movie*)malloc(sizeof(movie)*30);`
că tu ai alocat spațiu doar pentru 30 de pointeri către 30 de obiecte de tipul `movie`, pe când, fiecare dintre aceste 30 de obiecte conțin un pointer către un `char * name` <-- *ȘI ĂSTA* trebuie alocat!

Uită-te la exemplele pe care le-am făcut eu. De fiecare dată când aloc o structură care mai are un șir de .. ceva, orice, aloc și acel șir! (de obicei am denumit funcțiile de alocare `initialize ... `